<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:macro name="spot_ros2_control" params="interface_type has_arm">
        <!-- Currently implements a simple system interface that covers all joints of the robot.
        In the future, we could make different hardware interfaces for the body, arm, etc. -->
        <ros2_control name="SpotSystem" type="system">
            <hardware>
            <xacro:if value="${interface_type == 'mock'}">
                <plugin>mock_components/GenericSystem</plugin>
                <param name="mock_sensor_commands">false</param>
                <param name="state_following_offset">0.0</param>
            </xacro:if>
            <xacro:if value="${interface_type == 'spot-sdk'}">
                <plugin>spot_ros2_control/SpotHardware</plugin>
                <param name="example_param_hw_start_duration_sec">0</param>
                <param name="example_param_hw_stop_duration_sec">3.0</param>
                <param name="example_param_hw_slowdown">100</param>
            </xacro:if>
            </hardware>
            <!-- Add the legs -->
            <xacro:leg front_or_rear="front" left_or_right="left" />
            <xacro:leg front_or_rear="front" left_or_right="right" />
            <xacro:leg front_or_rear="rear" left_or_right="left" />
            <xacro:leg front_or_rear="rear" left_or_right="right" />
            <!-- Add the arm + the gripper if the robot has an arm -->
            <xacro:if value="${has_arm}">
                <xacro:arm/>
            </xacro:if>
        </ros2_control>
    </xacro:macro>

</robot>
